src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -91,11 +102,12 @@ class MovingWindow:
     ```
     """
 
-    def __init__(
+    def __init__(  # pylint: disable=too-many-arguments
         self,
         size: timedelta,
         resampled_data_recv: Receiver[Sample],
-        sampling_period: timedelta,
+        input_sampling_period: timedelta,
+        resampler_config: ResamplerConfig | None = None,
```

Comment:
Just a mental note I want to share. I was in doubt if creating the `ResamplerConfig`, as it added one more layer when creating a resampler (`Resampler(ResamplerConfig(...))` instead of just `Resampler(...)`) but now seeing it used, I think it was a huge win, it is very easy to use the resampler.

Without it it would be a nightmare to provide the full flexibility to configure it, having to repeat many very specific configuration options and forward them, and duplicate the documentation, or just give up and only take a resampling period, and lose all the resampler flexibility in the way.

I think we should apply this pattern in any class that can be created using some sort of non-trivial configuration.

Reply:
Well, I had a similar thought when Sahas showed the code in #270 and found it very convenient considering flexibility. I believe it will scale well with futures uses cases